limitador/src/storage/redis/counters_cache.rs

```diff
@@ -5,25 +5,84 @@ use crate::storage::redis::{
     DEFAULT_TTL_RATIO_CACHED_COUNTERS,
 };
 use moka::sync::Cache;
-use std::sync::Arc;
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicI64, Ordering};
+use std::sync::{Arc, Mutex};
 use std::time::{Duration, SystemTime};
+use tokio::select;
+use tokio::sync::Notify;
+use tokio::time::interval;
 
 pub struct CachedCounterValue {
     value: AtomicExpiringValue,
+    initial_value: AtomicI64,
     expiry: AtomicExpiryTime,
 }
 
+pub struct Batcher {
+    updates: Mutex<HashMap<Counter, Arc<CachedCounterValue>>>,
```

Comment:
This should eventually be replaced with a [concurrent HashMap](https://crates.io/crates/dashmap)